neuronunit/models/__init__.py

```diff
@@ -105,8 +113,10 @@ def set_lems_attrs(self, attrs):
                 for key2,value2 in value1.items():
                     node.attrib[key2] = value2
         tree.write(self.lems_file_path)
-
+    
     def run(self, rerun=None, **run_params):
+        #self.results = self._backend.local_run()
+        #This breaks NEURON backend
```

Comment:
@russelljjarvis 
If this breaks the `NEURONBackend`, how and where are you actually calling it's `local_run()`?  If `model.run()` doesn't call it, then what does?  It must eventually get called at some point to generate the results.

Reply:
It gets run in inject_square_current.

Reply:
I think it makes more sense for `inject_square_current` to be setting up the current injection, but for the simulation to not actually occur until `run` is invoked (which will ultimately call `local_run`), which will happen once some sort of output is requested (e.g. by something like `get_membrane_potential`).  Here is why: in the future we may do many things besides injecting square current, as a part of setting up the "simulation experiment" that is being done on the model.  We don't want to actually run the model every time we do one of these setup steps, only once at the end after the model is all setup.  

So `self.local_run()` should not be called in `inject_square_current`, but only in `run` (i.e. in the lines you commented out above).  Is there a reason why this would break the backend?

Reply:
I am unsure why it breaks it. I can try again, and design a unit test based on detecting the break.